typespec-extension/package.json

```diff
@@ -70,6 +71,13 @@
     "typescript": "~5.3.2",
     "@typescript-eslint/parser": "~6.12.0",
     "@typescript-eslint/eslint-plugin": "~6.12.0",
-    "eslint-plugin-deprecation": "~2.0.0"
+    "eslint-plugin-deprecation": "~2.0.0",
+    "@azure-tools/typespec-azure-core": "0.36.0",
+    "@azure-tools/typespec-client-generator-core": "0.36.0",
+    "@typespec/compiler": "0.50.0",
+    "@typespec/http": "0.50.0",
+    "@typespec/openapi": "0.50.0",
+    "@typespec/rest": "0.50.0",
+    "@typespec/versioning": "0.50.0"
```

Comment:
why do we need this? they should already be defined in `peerDependencies`?

Reply:
There are 2 primary motivations for listing the package in both devDependencies and peerDependencies.

1. To construct an installable emitter-package.json when the emitter depends on a dev version of typespec, we have to declare our dependency on the dev version of all of the peer dependencies:
    https://github.com/Azure/azure-sdk-for-java/pull/37850/files#diff-bac355930627a26bbb83d1f84d110b1e3ec82614739953f538daffc255e31d73
    ``` json
      "dependencies": {
        "@azure-tools/typespec-java": "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-js-test-autorest/npm/registry/@azure-tools/typespec-java/-/typespec-java-0.11.2-alpha.20231128.3.tgz"
      },
      "devDependencies": {
        "@typespec/openapi": "0.51.0-dev.1",
        "@typespec/compiler": "0.51.0-dev.9",
        "@typespec/versioning": "0.51.0-dev.6",
        "@typespec/rest": "0.51.0-dev.2",
        "@typespec/http": "0.51.0-dev.3",
        "@azure-tools/typespec-azure-core": "0.37.0-dev.6",
        "@azure-tools/typespec-client-generator-core": "0.37.0-dev.6"
      }
    ```

    The devDependency list in emitter-package json is constructed by comparing the peerDependency and devDependency list of the emitter's package.json file and including any package that's in both.

2. The specific version specified in devDependencies tells devs which version within the typically broad peerDependency range we're actually working with and is known to work with the module.

Reply:
One thought on generating the `devDependencies` on "emitter-package.json" in SDK repo is always use latest dev version (ignore the "devDependencies" here).

If both typespec-java and SDK preview is daily, these dev version should match.

Just a though. Feel free to disagree :-)